src/app/shared/components/template/components/youtube/youtube.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, OnInit } from "@angular/core";
+import { TemplateBaseComponent } from "../base";
+import { DomSanitizer, SafeResourceUrl } from "@angular/platform-browser";
+import {
+  getBooleanParamFromTemplateRow,
+  getStringParamFromTemplateRow,
+} from "src/app/shared/utils";
+import { TemplateTranslateService } from "../../services/template-translate.service";
+
+interface IYoutubeParams {
+  /** TEMPLATE PARAMETER: video_id */
+  videoId: string;
+  /** TEMPLATE PARAMETER: allow_full_screen. Default true */
+  allowFullScreen?: boolean;
+}
+
+@Component({
+  selector: "youtube",
+  templateUrl: "./youtube.component.html",
+  styleUrls: ["./youtube.component.scss"],
+})
+export class YoutubeComponent extends TemplateBaseComponent implements OnInit {
+  params: Partial<IYoutubeParams> = {};
+  src: SafeResourceUrl;
+
+  constructor(
+    private domSanitizer: DomSanitizer,
+    private templateTranslateService: TemplateTranslateService
+  ) {
+    super();
+  }
+
+  ngOnInit() {
+    this.getParams();
+    this.getYoutubeSrc();
+  }
+
+  private getParams() {
+    this.params.videoId =
+      this._row.value || getStringParamFromTemplateRow(this._row, "video_id", "");
+    this.params.allowFullScreen = getBooleanParamFromTemplateRow(
+      this._row,
+      "allow_fullscreen",
+      true
+    );
+  }
+
+  private getYoutubeSrc() {
+    const baseYoutubeUrl = `https://www.youtube.com/embed/${this.params.videoId}`;
+
+    // See https://developers.google.com/youtube/player_parameters
+    const youtubeQueryParams = {
+      // Favour white over red for more theme compatibility
+      color: "white",
+      // hide the fullscreen button if allow_fullscreen is false
+      fs: this.params.allowFullScreen ? "1" : "0",
+      // Attempt to set the player's interface language to match the app language
+      hl: this.templateTranslateService.app_language$.value,
```

Comment:
nit(non-blocking): nice idea to try and support the language, although if I think our `language` is more of a localisation code (country+lang, e.g. `tz_sw`), so I'm guessing for this to work would likely need to split the string and take the end characters.

Reply:
You're right, from [the docs](https://developers.google.com/youtube/player_parameters) it doesn't look like they support our (non-standard?) format, I've updated to try and extract the country code